### PR TITLE
compiler: Switch from aomp to clang for amd 

### PR DESCRIPTION
**Pull request by mloubout**

### CODE REVIEW

File: .github/workflows/pytest-gpu.yml [hunk 1]
```diff
@@ -46,31 +46,34 @@ jobs:
...
-          # https://github.com/ROCm-Developer-Tools/aomp/issues/397
-          test_files: "tests/test_gpu_openmp.py"
+          test_files: "tests/test_adjoint.py tests/test_gpu_common.py tests/test_gpu_openmp.py"
           tags: ["self-hosted", "amdgpu"]
           base: "devitocodes/bases:amd"
           test_drive_cmd: "rocm-smi"
           # Attach the AMD GPU devices `/dev` and add user to video and render (109 on wampa) group
           # Options from https://rocmdocs.amd.com/en/latest/ROCm_Virtualization_Containers/ROCm-Virtualization-&-Containers.html
           flags: "--network=host --device=/dev/kfd --device=/dev/dri --ipc=host --group-add video --group-add 109 --cap-add=SYS_PTRACE --security-opt seccomp=unconfined --rm --name testrun-amd"
+          mpi_flags: "--mca btl '^openib' -x UCX_TLS=sm,self,rocm_copy,rocm_ipc  --mca pml ucx"
```
Comment (FabioLuporini) [inline .github/workflows/pytest-gpu.yml:76]: Can we make them part of the dockerfile somehow?

File: docker/Dockerfile.cpu [hunk 2]
```diff
@@ -24,7 +24,7 @@ RUN apt-get install -y libgl1-mesa-glx
 ##############################################################
 FROM base as gcc
 
-RUN apt-get install -y mpich libmpich-dev 
+RUN apt-get install -y mpich libmpich-dev
```
Comment (FabioLuporini) [inline docker/Dockerfile.cpu:27]: extra space?

File: .github/workflows/pytest-gpu.yml [hunk 3]
```diff
@@ -46,31 +46,34 @@ jobs:
           pytest-gpu-acc-nvidia,
           pytest-gpu-omp-amd
         ]
+        test_examples: ["examples/seismic/tti/tti_example.py examples/seismic/acoustic/acoustic_example.py examples/seismic/viscoacoustic/viscoacoustic_example.py examples/seismic/viscoelastic/viscoelastic_example.py examples/seismic/elastic/elastic_example.py"]
+
         include:
         - name: pytest-gpu-omp-nvidia
           test_files: "tests/test_adjoint.py tests/test_gpu_common.py tests/test_gpu_openmp.py"
```
Comment (FabioLuporini) [inline .github/workflows/pytest-gpu.yml:53]: Now that they are the same for all builds, we can pull this variable outside the `include` scope, and maybe put it right under the newly introduced `test_examples` ?

Comment (mloubout) [reply .github/workflows/pytest-gpu.yml:53]: They are not the openacc build is different

File: .github/workflows/pytest-gpu.yml (hunk 1 above)
Comment (mloubout) [reply .github/workflows/pytest-gpu.yml:76]: We could be this makes it slightly easier for ci

Comment (FabioLuporini) [reply .github/workflows/pytest-gpu.yml:76]: but OK why do these options only matter for CI and not for the end users?

Comment (mloubout) [reply .github/workflows/pytest-gpu.yml:76]: No it does too and should be moved to the docker at some point, but I would need to re-build the docker separately , test it, then restart that PR so it's not very practical that's all

File: docker/Dockerfile.cpu (hunk 2 above)
Comment (mloubout) [reply docker/Dockerfile.cpu:27]: No removed extra space

File: .github/workflows/pytest-gpu.yml (hunk 1 above)
Comment (mloubout) [reply .github/workflows/pytest-gpu.yml:76]: As mentionned in slack these should go in the `mpiexec.conf`

File: .github/workflows/pytest-gpu.yml (hunk 3 above)
Comment (FabioLuporini) [reply .github/workflows/pytest-gpu.yml:53]: ah right !

File: .github/workflows/pytest-gpu.yml (hunk 1 above)
Comment (mloubout) [reply .github/workflows/pytest-gpu.yml]: fixed

**Pull request merged**